egs2/ami/asr1/local/data.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+. ./db.sh || exit 1;
+
+stage=0
+stop_stage=10
+SECONDS=0
+
+# You may set 'mic' to:
+#  ihm [individual headset mic- the default which gives best results]
+#  sdm1 [single distant microphone- the current script allows you only to select
+#        the 1st of 8 microphones]
+#  mdm8 [multiple distant microphones-- currently we only support averaging over
+#       the 8 source microphones].
+# ... by calling run.sh as, for example,
+# ./run.sh --mic sdm1
+# ./run.sh --mic mdm8
+mic=ihm
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+
+if [ ! -e "${AMI}" ]; then
+    log "Fill the value of 'AMI' of db.sh"
+    exit 1
+fi
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+base_mic=${mic//[0-9]/} # sdm, ihm or mdm
+nmics=${mic//[a-z]/} # e.g. 8 for mdm8.
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    log "data stage 1: Data Download"
+    if [ -d ${AMI} ] && ! touch ${AMI}/.foo 2>/dev/null; then
+	log "$0: directory $AMI seems to exist and not be owned by you."
+	log " ... Assuming the data does not need to be downloaded.  Please use --stage 2."
+	exit 1
+    fi
+    if [ -e data/local/downloads/wget_${mic}.sh ]; then
+	log "data/local/downloads/wget_$mic.sh already exists, better quit than re-download... (use --stage N)"
+	exit 1
+    fi
+    local/ami_download.sh ${mic} ${AMI}
+fi
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+    ### Task dependent. You have to make data the following preparation part by yourself.
+    ### But you can utilize Kaldi recipes in most cases
+    log "data stage 2: Data preparation"
+
+    # common data prep
+    if [ ! -d data/local/downloads ]; then
+	local/ami_text_prep.sh data/local/downloads
+    fi
+
+    # beamforming
+    if [ "$base_mic" == "mdm" ]; then
+	PROCESSED_AMI_DIR=${PWD}/beamformed
+	if [ -z ${BEAMFORMIT} ]; then
+	    export BEAMFORMIT=${KALDI_ROOT}/tools/BeamformIt
+	fi
+	export PATH=${PATH}:${BEAMFORMIT}
+	! hash BeamformIt && log "Missing BeamformIt, run 'cd ../../../tools/kaldi/tools; extras/install_beamformit.sh; cd -;'" && exit 1
+	local/ami_beamform.sh --cmd "${train_cmd}" --nj 20 ${nmics} ${AMI} ${PROCESSED_AMI_DIR}
+    else
+	PROCESSED_AMI_DIR=${AMI}
+    fi
+    local/ami_${base_mic}_data_prep.sh ${PROCESSED_AMI_DIR} ${mic}
+    # data augmentation
+    utils/perturb_data_dir_speed.sh 0.9 data/${mic}/train_orig data/ihm_tmp1
+    utils/perturb_data_dir_speed.sh 1.0 data/${mic}/train_orig data/ihm_tmp2    
+    utils/perturb_data_dir_speed.sh 1.1 data/${mic}/train_orig data/ihm_tmp3
```

Comment:
Change to use --speed_perturb_factors
https://github.com/espnet/espnet/blob/24b07c73e0c3b98177d376cb7e0e5c936ed78a53/egs2/aishell/asr1/run.sh#L19-L36